### DISCUSSION TITLE
Cloning private repo with a Github App private key

### DISCUSSION THREAD

**Discussion by camilosampedro** [community/community#24575] [Apps, API and Webhooks]
<p>Hello!</p>
<p>I was wondering if it is possible to clone a private repository using a Github App private key.</p>
<p>Is it possible?</p>
<p>If so, do you know which permissions it requires to do so?</p>
<p>I tried giving it “Contents” read-only permissions, but it still cannot clone it…</p>
<p>The way I test it is by using:</p>
<pre><code>ssh-agent bash -c 'ssh-add {{private_key_file.pem}}; git clone git@github.com:{{username}}/{{repo}}.git'
</code></pre>

**Comment by kevcodez** [comment, marked as answer]
<p>Did you find a solution?</p>
<p>I am trying to clone private repositories with my Github App aswell.</p>
<p>I just found a way that works for me using the command line.</p>
<p><a href="https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/" rel="nofollow noopener">https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/</a> at the very bottom</p>
<pre><code>git clone https://x-access-token:&lt;token&gt;@github.com/owner/repo.git
</code></pre>
<p>I am retrieving a user token using a POST request to <a href="https://api.github.com/app/installations/%24installationId/access_tokens" rel="nofollow noopener">https://api.github.com/app/installations/$installationId/access_tokens</a>.</p>

**Reply by joeblubaugh** [reply]
Unfortunately this is no longer supported, even though the official documentation claims it is.

**Reply by joshuamurari-Contoso-MCAPS** [reply]
@joeblubaugh , have you found any other solution. This also does not work for me at the moment.

**Reply by btello** [reply]
@joshuamurari-Contoso-MCAPS For my use case I was able to use a Deploy Key defined on the repository I needed to clone https://docs.github.com/en/authentication/connecting-to-github-with-ssh/managing-deploy-keys#deploy-keys.

**Reply by smeeklai** [reply]
Still worked for me

**Reply by daviewales** [reply]
This worked for me, as did:

```
git clone https://git:<token>@github.com/owner/repo.git
```

**Comment by vishal-kapse** [comment]
Cloning a private repo using GitHub App can be achieved by below approach

1. Generate a JWT for your app using the key file
[Generating a JSON Web Token (JWT)](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app#generating-a-json-web-token-jwt)

2. Generate a Installation Access token
[Generating an installation access token](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation#generating-an-installation-access-token)

3. Use the token generated to authenticate
[Authenticating with an installation access token](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation#authenticating-with-an-installation-access-token)

Done !!

**Reply by saarw-opti** [reply]
For some reason, I am getting remote: repository not found, and I've opened the necessary permissions in the GitHub application.

**Reply by deanmalan** [reply]
I'm using the [`actions/create-github-app-token`](https://github.com/actions/create-github-app-token) and needed to specify the [`repositories`](https://github.com/actions/create-github-app-token?tab=readme-ov-file#repositories) and/or [`owner`](https://github.com/actions/create-github-app-token?tab=readme-ov-file#owner) input fields to grant access to other repositories.

**Reply by niksauer** [reply]
@saarw-opti Be sure to check that you have granted the `contents:read` repository permission. This was missing for me.

**Comment by shamimrahat7** [comment]
> Hello!
> 
> I was wondering if it is possible to clone a private repository using a Github App private key.
> 
> Is it possible?
> 
> If so, do you know which permissions it requires to do so?
> 
> I tried giving it “Contents” read-only permissions, but it still cannot clone it…
> 
> The way I test it is by using:
> 
> ```
> ssh-agent bash -c 'ssh-add {{private_key_file.pem}}; git clone git@github.com:{{username}}/{{repo}}.git'
> ```
